src/backend/distributed/utils/connection_management.c

```diff
@@ -0,0 +1,635 @@
+/*-------------------------------------------------------------------------
+ *
+ * connection_management.c
```

Comment:
I'm guessing we'll at least have functions for dealing with many connections. Could also be part of executor/ maybe.